src/config.rs

```diff
@@ -312,7 +333,7 @@ impl Config {
                 namespace if namespace.is_case(Case::UpperFlat) => {
                     packages::Namespace::Namespace(namespace.to_string())
                 }
-                namespace => packages::Namespace::Namespace(namespace.to_string().to_case(Case::Pascal)),
+                namespace => packages::Namespace::Namespace(namespace_from_package_name(namespace)),
```

Comment:
There is another use of `to_case(Case::Pascal)` on line 348, but actually I think you can simplify this match quite a lot and always process namespace whatever the initial case with this new function.